### PR TITLE
DDF-3506 Improve OWASP scanning

### PR DESCRIPTION
**Pull request by ahoffer**
#### What does this PR do?
- Reduce unnecessary OWASP checks for CI
- Update the dependency check plugin version to 3.1.1
- Use maven plugin management to set default properties for dependency check plugin
#### Who is reviewing it? 
@mcalcote
@vinamartin
@brianfelix
@LinkMJB
@oconnormi

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)

@clockard
@rzwiefel

#### How should this be tested? (List steps with links to updated documentation)

`mvn install -P owasp -DskipStatic=true -DskipTests=true`

#### Any background context you want to provide?
No.
#### What are the relevant tickets?
DDF-3506

#### Checklist:
- [ N/A] Documentation Updated
- [N/A ] Update / Add Unit Tests
- [N/A ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an inline code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by ahoffer** [comment]
build now

File: pom.xml [hunk 1]
```diff
@@ -328,6 +328,47 @@
...
+                        <assemblyAnalyzerEnabled>false</assemblyAnalyzerEnabled>
+                        <!--Analyzes Ruby Gemfile.lock files, not OSGi bundles-->
+                        <bundleAuditAnalyzerEnabled>false</bundleAuditAnalyzerEnabled>
+                        <!--We don't use CMake, CocoaPods, AutoConf, or Ruby-->
+                        <cmakeAnalyzerEnabled>false</cmakeAnalyzerEnabled>
+                        <cocoapodsAnalyzerEnabled>false</cocoapodsAnalyzerEnabled>
+                        <autoconfAnalyzerEnabled>false</autoconfAnalyzerEnabled>
+                        <rubygemsAnalyzerEnabled>false</rubygemsAnalyzerEnabled>
+                        <!--NSP is used separately -->
+                        <nspAnalyzerEnabled>false</nspAnalyzerEnabled>
```
Comment (clockard) [inline pom.xml]: This is a duplicate of line 341

**Comment by ahoffer** [comment]
build now
Rebuilding after rebasing

**Comment by ahoffer** [comment]
build now

**Comment by vinamartin** [comment]
```02:50:42 [ERROR] Failed to execute goal org.owasp:dependency-check-maven:3.1.1:check (default) on project platform-email-impl:
02:50:42 [ERROR] 
02:50:42 
[ERROR] One or more dependencies were identified with vulnerabilities that have a CVSS score greater than or equal to '7.0':
02:50:42 [ERROR] 
02:50:42 [ERROR] simple-java-mail-4.1.3.jar/META-INF/maven/com.github.bbottema/emailaddress-rfc2822/pom.xml: CVE-2015-7686```

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

java.lang.Exception: Boom!!!
	at ddf.test.itests.catalog.TestMessageBroker$1.lambda$configure$0(TestMessageBroker.java:300) ~[?:?]

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.6.CODICE:yarn (yarn install) on project ui: Failed to run task: 'yarn install --mutex network' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)

**Pull request merged**